src/Chapter3.hs

```diff
@@ -912,6 +976,20 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold Int
+
+instance Append Gold where
+  append (Gold g1) (Gold g2) = Gold (g1 + g2)
+instance Append [a] where
+  append = (++)
+
+instance (Append a) => Append (Maybe a) where
+  append (Just x) Nothing = Just x
+  append Nothing (Just x)  = Just x
+  append (Just x) (Just y) = Just $ append x y
+  append Nothing Nothing = Nothing
```

Comment:
This is an absolutely correct solution!

However, we can notice that these 3 cases here could be reduced to 2!
Let's see what we can unite in here.

If some of the arguments are `Nothing`, you actually always return the second one! So you need to inspect only two cases: `Nothing` in the first position, and `Nothing` in the second one 🙂